core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java

```diff
@@ -354,7 +369,13 @@ public static Object getMeasureDefaultValueByType(ColumnSchema columnSchema,
       // then setting measure exists is true
       // otherwise adding a default value of a measure
       for (CarbonMeasure carbonMeasure : currentBlockMeasures) {
-        if (carbonMeasure.getColumnId().equals(queryMeasure.getMeasure().getColumnId())) {
+        // If it is unmanaged table just check the column names, no need to validate column id as
+        // multiple sdk's output placed in a single folder doesn't have same column ID but can
+        // have same column name
+        if (carbonMeasure.getColumnId().equals(queryMeasure.getMeasure().getColumnId()) ||
+            ((queryModel != null) && (queryModel.getTable().getTableInfo().isUnManagedTable()) &&
```

Comment:
remove null check and add method isColumnMatches

Reply:
ok

Reply:
removed null check. cannot call from a method as one is for measure type and other is a dimension type.